### PR TITLE
[FLOC-4236] reorganise the golang files in repository

### PR DESCRIPTION
**Pull request by mdevilliers**
Move the datalayer and cmd packages to their new home in pkg.

### CODE REVIEW

File: dvol_test.go [hunk 1]
```diff
@@ -1 +0,0 @@
-package main
```
Comment (sarum90) [inline dvol_test.go]: Is this now an empty file, or is it deleted? If not deleted, could it be deleted?

**Comment by sarum90** [comment]
What I reviewed for:

Things were moved into a folder named 'pkg'.

References to those things had 'pkg' added to the path without typos.

Looks good to me. Merge when Travis is happy.

I _might_ do some code grepping to double check you caught all references, but don't hold up merging on that.

**Comment by sarum90** [comment]
Confirmed with grep that you got all references to "/cmd" and to "/datalayer" :+1:

File: dvol_test.go (hunk 1 above)
Comment (mdevilliers) [inline dvol_test.go]: Yes it is deleted.

**Comment by mdevilliers** [comment]
Thanks for the review.

File: dvol_test.go (hunk 1 above)
Comment (sarum90) [inline dvol_test.go]: It's probably bad that I didn't know that was how gitHub showed that... we probably need to delete more flocker code...

**Pull request merged**